modules/ghlint-rules/src/main/kotlin/net/twisterrob/ghlint/rules/InvalidExpressionUsageRule.kt

```diff
@@ -0,0 +1,85 @@
+package net.twisterrob.ghlint.rules
+
+import net.twisterrob.ghlint.model.ActionStep
+import net.twisterrob.ghlint.model.WorkflowStep
+import net.twisterrob.ghlint.rule.Example
+import net.twisterrob.ghlint.rule.Issue
+import net.twisterrob.ghlint.rule.Reporting
+import net.twisterrob.ghlint.rule.report
+import net.twisterrob.ghlint.rule.visitor.ActionVisitor
+import net.twisterrob.ghlint.rule.visitor.VisitorRule
+import net.twisterrob.ghlint.rule.visitor.WorkflowVisitor
+
+public class InvalidExpressionUsageRule : VisitorRule, ActionVisitor, WorkflowVisitor {
+
+	override val issues: List<Issue> = listOf(InvalidExpression)
+
+	override fun visitWorkflowUsesStep(reporting: Reporting, step: WorkflowStep.Uses) {
+		super.visitWorkflowUsesStep(reporting, step)
+
+		if (step.uses.uses.containsGitHubExpression()) {
+			reporting.report(InvalidExpression, step) {
+				"${it} contains a GitHub expression in the `uses` field."
+			}
+		}
+	}
+
+	override fun visitActionUsesStep(reporting: Reporting, step: ActionStep.Uses) {
+		super.visitActionUsesStep(reporting, step)
+
+		if (step.uses.uses.containsGitHubExpression()) {
+			reporting.report(InvalidExpression, step) {
+				"${it} contains a GitHub expression in the `uses` field."
+			}
+		}
+	}
+
+	private companion object {
+		const val uses = "actions/checkout@\${{ github.sha }}"
+
+		val InvalidExpression = Issue(
+			id = "InvalidExpressionUsage",
+			title = "Expressions should not be used in uses field.",
+			description = """
+				> GitHub Action Expressions can be used to programmatically access context variables in workflow files.
+				> -- [About expressions](https://docs.github.com/en/actions/learn-github-actions/expressions)
+				
+				However, they cannot be used within the uses field. 
+				GitHub will report an error: 
+				> The workflow is not valid. `.github/workflows/???.yml` (Line: ?, Col: ?):
+				> A template expression is not allowed in this context
+			""".trimIndent(),
```

Comment:
`title` is what shows on PRs as the first sentence. I swapped it around so it's more consistent with other issues.

```suggestion
			title = "Step `uses` must not contain expressions.",
			description = """
				> GitHub Action Expressions can be used to programmatically access context variables in workflow files.
				> -- [About expressions](https://docs.github.com/en/actions/learn-github-actions/expressions)
				
				However, they cannot be used within `steps[*].uses`.
				GitHub will report an error: 
				> The workflow is not valid. `.github/workflows/???.yml` (Line: ?, Col: ?):
				> A template expression is not allowed in this context
			""".trimIndent(),
```
(Tried to also remove the word "field", checked the docs for what it is, but they also just avoid qualifying it and rely on syntax highlight (`uses` or `uses:`) to distinguish. 🤷‍♂️)